src/vs/workbench/contrib/terminal/browser/terminalLinkHandler.ts

```diff
@@ -89,15 +89,38 @@ export class TerminalLinkHandler {
 		// Matches '+++ b/src/file1', capturing 'src/file1' in group 1
 		this._gitDiffPostImagePattern = /^\+\+\+ b\/(\S*)/;
 
-		this._tooltipCallback = (e: MouseEvent) => {
+		this._tooltipCallback = (e: MouseEvent, uri: string, location: IViewportRange) => {
 			if (!this._widgetManager) {
 				return;
 			}
+
+			// Get the row bottom up
+			let offsetRow = this._xterm.rows - location.start.row + 1;
+			let useTopPosition = false;
+
+			// Handle a link on the top row
+			if (location.start.row === 1) {
+				offsetRow--;
+				useTopPosition = true;
+			}
+
 			if (this._configHelper.config.rendererType === 'dom') {
-				const target = (e.target as HTMLElement);
-				this._widgetManager.showMessage(target.offsetLeft, target.offsetTop, this._getLinkHoverString());
+				const font = this._configHelper.getFont();
+				const charWidth = font.charWidth;
+				const charHeight = font.charHeight;
+
+				const leftPosition = (location.start.col - 1) * (charWidth! + (font.letterSpacing / window.devicePixelRatio));
+				const bottomPosition = offsetRow * (charHeight! * font.lineHeight);
+
+				this._widgetManager.showMessage(leftPosition, bottomPosition, this._getLinkHoverString(), useTopPosition);
```

Comment:
Got it in this state with the dom renderer when zoomed in (ctrl/cmd++)

![image](https://user-images.githubusercontent.com/2193314/67436776-1b1f7980-f5a4-11e9-9e03-fa2f0c163217.png)